python/lsst/daf/butler/script/queryCollections.py

```diff
@@ -28,20 +28,40 @@
 from __future__ import annotations
 
 from collections.abc import Iterable
+from fnmatch import fnmatch
 from typing import Literal
 
-from astropy.table import Table
+from astropy.table import Column, Table, hstack, vstack
 
 from .._butler import Butler
 from .._butler_collections import CollectionInfo
 from .._collection_type import CollectionType
 
 
+def _parseDatasetTypes(dataset_types: frozenset[str] | list[str] | None) -> list[str]:
+    """Parse dataset types from a collection info object or a list of strings.
+
+    Parameters
+    ----------
+    dataset_types : `frozenset`[`str`] | `list`[`str`] | `None`
+        The dataset types to parse. If `None`, an empty list is returned.
+        If a `frozenset` or `list` is provided, it is returned as a list.
+
+    Returns
+    -------
+    dataset_types : `list`[`str`]
+        The parsed dataset types.
+    """
+    return [""] if not dataset_types else list(dataset_types)
+
+
 def _getTable(
     repo: str,
     glob: Iterable[str],
     collection_type: Iterable[CollectionType],
     inverse: bool,
+    show_dataset_types: bool = False,
+    exclude_dataset_types: Iterable[str] = [],
```

Comment:
Generally python doesn't really like this approach because you get the same list each time. Here it doesn't really matter since the list is not modified but to avoid a picky reviewer/linter worrying about it in the future:
```suggestion
    exclude_dataset_types: Iterable[str] | None = None,
```

and then
```python
excluded = exclude_dataset_types if exclude_dataset_types is not None else []
```
(although it looks like the code does `if exclude_dataset_types:` so should work fine with None.)

Reply:
I've implemented this everywhere it appears, and modified the doc strings to match.

My testing shows that the `excluded = ...` logic was not necessary, for the reasons you suggest, so I've not included that.